src/language.rs

```diff
@@ -30,161 +30,148 @@ pub enum SupportedLanguageName {
     Lua,
 }
 
-impl SupportedLanguageName {
-    pub fn get_language(&self) -> SupportedLanguage {
-        *ALL_SUPPORTED_LANGUAGES_BY_SUPPORTED_LANGUAGE_NAME
-            .get(self)
-            .unwrap()
-    }
-}
+pub type BySupportedLanguage<T> = [T; 22];
+
+impl<T> Index<SupportedLanguage> for BySupportedLanguage<T> {
+    type Output = T;
 
-#[derive(Copy, Clone, Debug)]
-pub struct SupportedLanguage {
-    pub language: Language,
-    pub name: SupportedLanguageName,
-    pub name_for_ignore_select: &'static str,
+    fn index(&self, index: SupportedLanguage) -> &Self::Output {
+        &self[index as usize]
+    }
 }
 
-impl PartialEq for SupportedLanguage {
-    fn eq(&self, other: &Self) -> bool {
-        self.language == other.language
+impl From<usize> for SupportedLanguage {
+    fn from(value: usize) -> Self {
+        match value {
+            value if value == Self::Rust as usize => Self::Rust,
+            value if value == Self::Typescript as usize => Self::Typescript,
+            value if value == Self::Javascript as usize => Self::Javascript,
+            value if value == Self::Swift as usize => Self::Swift,
+            value if value == Self::ObjectiveC as usize => Self::ObjectiveC,
+            value if value == Self::Toml as usize => Self::Toml,
+            value if value == Self::Python as usize => Self::Python,
+            value if value == Self::Ruby as usize => Self::Ruby,
+            value if value == Self::C as usize => Self::C,
+            value if value == Self::Cpp as usize => Self::Cpp,
+            value if value == Self::Go as usize => Self::Go,
+            value if value == Self::Java as usize => Self::Java,
+            value if value == Self::CSharp as usize => Self::CSharp,
+            value if value == Self::Kotlin as usize => Self::Kotlin,
+            value if value == Self::Elisp as usize => Self::Elisp,
+            value if value == Self::Elm as usize => Self::Elm,
+            value if value == Self::Dockerfile as usize => Self::Dockerfile,
+            value if value == Self::Html as usize => Self::Html,
+            value if value == Self::TreeSitterQuery as usize => Self::TreeSitterQuery,
+            value if value == Self::Json as usize => Self::Json,
+            value if value == Self::Css as usize => Self::Css,
+            value if value == Self::Lua as usize => Self::Lua,
+            _ => unreachable!(),
```

Comment:
Obviously this is not technically true (that no `usize` argument could trigger this code path) but I think we'd be ok panicking if we were trying to convert a non-"known good" `usize` back into the enum, so using `From` (instead of eg `TryFrom`) seems more ergonomically desirable